src/main.rs

```diff
@@ -672,6 +679,79 @@ impl Context {
         Ok(())
     }
 
+    fn open_blog(&mut self) -> Result<(), Error> {
+        // We rely on the blog variables not being set in production to disable
+        // blogging on the actual release date.
+        if self.config.channel != Channel::Stable {
+            eprintln!("Skipping blogging -- not on stable");
+            return Ok(());
+        }
+
+        let mut github = if let Some(github) = self.config.github() {
+            github
+        } else {
+            eprintln!("Skipping blogging - GitHub credentials not configured");
+            return Ok(());
+        };
+        let mut discourse = if let Some(discourse) = self.config.discourse() {
+            discourse
+        } else {
+            eprintln!("Skipping blogging - Discourse credentials not configured");
+            return Ok(());
+        };
+        let repository_for_blog = if let Some(repo) = &self.config.blog_repository {
+            repo.as_str()
+        } else {
+            eprintln!("Skipping blogging - blog repository not configured");
+            return Ok(());
+        };
+
+        let version = self.current_version.as_ref().expect("has current version");
+        let internals_contents =
+            if let Some(contents) = self.config.blog_contents(version, &self.date, false, None) {
+                contents
+            } else {
+                eprintln!("Skipping internals - insufficient information to create blog post");
+                return Ok(());
+            };
+
+        let announcements_category = 18;
+        let internals_url = discourse.create_topic(
+            announcements_category,
+            &format!("Rust {} pre-release testing", version),
+            &internals_contents,
+        )?;
+        let blog_contents = if let Some(contents) =
+            self.config
+                .blog_contents(version, &self.date, true, Some(&internals_url))
+        {
+            contents
+        } else {
+            eprintln!("Skipping blogging - insufficient information to create blog post");
+            return Ok(());
+        };
+
+        // Create a new branch so that we don't need to worry about the file
+        // already existing. In practice this *could* collide, but after merging
+        // a PR branches should get deleted, so it's very unlikely.
+        let name = format!("automation-{:x}", rand::random::<u32>());
+        let mut token = github.token(repository_for_blog)?;
+        let master_sha = token.get_ref("heads/master")?;
```

Comment:
I'd put the `master` here in a constant, so that we don't have to dig for it when we switch to `main` as the main branch.